metricbeat/helper/conversion.go

```diff
@@ -0,0 +1,85 @@
+/*
+The conversion functions take a key and a map[string]string. First it checks if the key exists and logs and error
+if this is not the case. Second the conversion to the type is done. In case of an error and error is logged and the
+default values is returned. This guarantees that also if a field is missing or is not defined, still the full metricset
+is returned.
+*/
+package helper
+
+import (
+	"strconv"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// ToBool converts value to bool. In case of error, returns false
+func ToBool(key string, data map[string]string) bool {
+
+	exists := checkExist(key, data)
+	if !exists {
+		logp.Err("Key does not exist in in data: %s", key)
+		return false
+	}
+
+	value, err := strconv.ParseBool(data[key])
+	if err != nil {
+		logp.Err("Error converting param to bool: %s", key)
+		return false
+	}
+
+	return value
+}
+
+// ToFloat converts value to float64. In case of error, returns 0.0
+func ToFloat(key string, data map[string]string) float64 {
+
+	exists := checkExist(key, data)
+	if !exists {
+		logp.Err("Key does not exist in in data: %s", key)
+		return 0.0
+	}
+
+	value, err := strconv.ParseFloat(data[key], 64)
+	if err != nil {
+		logp.Err("Error converting param to float: %s", key)
+		value = 0.0
+	}
+
+	return value
+}
+
+// ToInt converts value to int. In case of error, returns 0
+func ToInt(key string, data map[string]string) int {
```

Comment:
Shouldn't we work on int64 by default (long in Elasticsearch)? I think longs will be pretty common in these datasets.

Btw, I recently learned that there's not much benefit in declaring numbers as integers in Elasticsearch, since integers are actually longs behind the scene (i.e. they take the exact same storage space).